### PR TITLE
Reed-Solomon Error Correction Coding

### PR DESCRIPTION
**Pull request by andrewray**
Software library and HardCaml hardware design.

### CODE REVIEW

File: packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/descr [hunk 1]
```diff
@@ -0,0 +1,6 @@
+HardCaml implementation of Reed-Solomon error correction coding.
+
+The HardCamlReedsolomon library provides the hardware design.
+
+The executables harcaml-rsencode and hardcaml-rsdecode provide
```
Comment (avsm) [inline packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/descr]: s/harcaml/hardcaml

File: packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/files/hardcaml-reedsolomon.install [hunk 2]
```diff
@@ -0,0 +1,4 @@
+bin: [
+"hcrsencode.native" {"hardcaml-rsencode"}
+"hcrsdecode.native" {"hardcaml-rsdecode"}
+]
```
Comment (avsm) [inline packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/files/hardcaml-reedsolomon.install]: You can put these directly in your source repo, btw (so it works with local `opam pin` workflows as well)

File: packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/descr (hunk 1 above)
Comment (andrewray) [inline packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/descr]: ta - fixed on latest commit

File: packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/files/hardcaml-reedsolomon.install (hunk 2 above)
Comment (andrewray) [inline packages/hardcaml-reedsolomon/hardcaml-reedsolomon.0.1/files/hardcaml-reedsolomon.install]: also done

**Pull request merged**